### ISSUE TITLE
Issue with memory persistence, messages getting redelivered

### ISSUE THREAD

**Issue by Debdutto** [mcollina/aedes#33]
So, I'm using aedes for testing so that we can shift to aedes from mosca. I have two node processes, one serving as the receiver and another one serving as the sender.

I'm sending around 2000 messages to the receiver. All messages are getting delivered successfully. But, if I'm disconnecting the receiver and connecting again, messages are getting delivered again in chunks following which the server crashes with the following error stack.

```
/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/lib/handlers/subscribe.js:33
  topicActions.call(this, sub, done)
               ^

RangeError: Maximum call stack size exceeded
    at SubscribeState.doSubscribe (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/lib/handlers/subscribe.js:33:16)
    at makeCall (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/node_modules/fastseries/series.js:113:10)
    at SubscribeState.ResultsHolder.release (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/node_modules/fastseries/series.js:94:9)
    at work (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/node_modules/fastfall/fall.js:121:25)
    at sendRetained (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/lib/handlers/subscribe.js:95:5)
    at SubscribeState.subTopic (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/lib/handlers/subscribe.js:90:5)
    at work (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/node_modules/fastfall/fall.js:105:23)
    at Aedes.authorizeSubscribe (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/aedes.js:281:3)
    at SubscribeState.authorize (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/lib/handlers/subscribe.js:38:17)
    at Holder.work (/Users/debduttochakraborty/Sites/TestCode/proxy-server/node_modules/aedes/node_modules/fastfall/fall.js:105:23)
```

**Comment by mcollina** [comment]
Can you please send me an example client that is triggering the issue? How
can I reproduce?
Il giorno mer 17 feb 2016 alle 20:24 Debdutto Chakraborty <
notifications@github.com> ha scritto:

**Comment by Debdutto** [comment]
Here's the code to reproduce the issue. I've also tried with different concurrency values also. I'm using the default in-memory persistence for this. Running on a Mac Pro, i5,  8GB RAM

https://gist.github.com/Debdutto/acca3b8a8fe8b6dff438

**Comment by mcollina** [comment]
It is a known issue. The redis/level persistences do not have this problem, and memory is highly discouraged for heavy loads (because you might end up filling up your memory anyway).

A quick solution is to increase the stack size with `node --stack_size NUM`.

You might want to send a PR to https://github.com/mcollina/aedes-persistence.

**Comment by Debdutto** [comment]
Okay. Thanks. I was just trying it out. Will anyway have to move to Redis or Aerospike.

**Comment by mcollina** [comment]
Reopening, it needs a fix :)

**Comment by mcollina** [comment]
Fixed in aedes-persistence 1.2.4.

**Comment by mcollina** [comment]
This solves the problem on my side :).
